docs: add semantic line break guidance to style conventions

imatwawana opened the pull request.

josmperez commented:
What you've written is fine and accurately conveys the decision made by the style council. However, as I recall this decision didn't attract a lot of discussion of potential pros/cons, and now that I see this PR I can think of some reasons to hit pause. First, is anyone really asking for this? Does this solve any problem that writers are having now? Second, are we now expected to go through all existing docs and apply SLBs? Is this worth the investment of time and money? If we don't, are we going to have ugly docs with a mix of SLB and non-SLB content? Third, I don't think the PR accurately represents that we follow SEMBR.org, because we currently don't, and these guidelines are much more extensive than anything the style council discussed.

Finally, my personal opinion has changed since the style council after reading SEMBR.org's website. I think SLB should probably be a decision left up to individual writers - just one okay way of writing.

imatwawana commented:
We landed at this being a recommendation vs a hard rule so that if someone wants guidance, there's something, but it's not a requirement. Not sure if that addresses some of your feedback. Maybe we can look at phrasing and how we make sure it's understood as a recommendation vs a rule, but ultimately, there's been a discussion, a poll, a clear outcome from the poll, and implementation work has been started.

josmperez commented:
> We landed at this being a recommendation vs a hard rule so that if someone wants guidance, there's something, but it's not a requirement. Not sure if that addresses some of your feedback. Maybe we can look at phrasing and how we make sure it's understood as a recommendation vs a rule, but ultimately, there's been a discussion, a poll, a clear outcome from the poll, and implementation work has been started. I don't know that it's a reasonable ask at this point to rehash it and I don't know how we avoid getting into this situation again if every thing is still up for debate once we get to this stage.

As I stated, I have no issue with individual writers using SLBs. But the PR as written states that Grafana recommends following the SLB guidance of sembr.org, which to be perectly clear is:

Semantic Line Breaks Specification (SemBr)
The key words “must”, “must not”, “required”, “shall”, “shall not”, “should”, “should not”, “recommended”, “may”, and “optional” in this document are to be interpreted as described in [RFC 2119](https://www.ietf.org/rfc/rfc2119.txt).

1. Text written as plain text or a compatible markup language may use semantic line breaks.
2. A semantic line break must not alter the final rendered output of the document.
3. A semantic line break should not alter the intended meaning of the text.
4. A semantic line break must occur after a sentence, as punctuated by a period (.), exclamation mark (!), or question mark (?).
5. A semantic line break should occur after an independent clause as punctuated by a comma (,), semicolon (;), colon (:), or em dash (—).
6. A semantic line break may occur after a dependent clause in order to clarify grammatical structure or satisfy line length constraints.
7. A semantic line break is recommended before an enumerated or itemized list.
8. A semantic line break may be used after one or more items in a list in order to logically group related items or satisfy line length constraints.
9. A semantic line break must not occur within a hyphenated word.
10. A semantic line break may occur before and after a hyperlink.
11. A semantic line break may occur before inline markup.
12. A maximum line length of 80 characters is recommended.
13. A line may exceed the maximum line length if necessary, such as to accommodate hyperlinks, code elements, or other markup.

All of these details were never discussed as a group, and most of us probably weren't aware of them (I wasn't).

imatwawana commented:
We wanted to provide a resource and a recommendation. The words in this PR are not in stone. Some folks on the team do follow those general guidelines, but again, this isn't meant to be hard and fast language. If you can suggest a possible edit to the language, I think that might be more productive.

chri2547 reviewed (approved):
I don't think it's necessary to discuss this any further. What Isabel has written looks perfectly reasonable to me.

GrafanaWriter commented on docs/sources/write/style-guide/style-conventions/index.md:
```diff
@@ -219,3 +219,23 @@ On the published page, this warning renders as follows:
+
+## Semantic line breaks
+
+We use [these general rules](https://sembr.org/) for adding line breaks in your writing, but we recommend starting a new line after each sentence to make the text easier to review and edit.
```
Given that we don't ourselves follow these rules in our documentation, may I suggest we soften this a bit:

```suggestion
The [Semantic Line Breaks organization](https://sembr.org/) suggests adding line breaks in your writing. We find this can help with reviews.

Note that we don't require that this suggestion be followed. We share the Semantic Line Breaks rules so that you can discuss with your review team and adapt to your circumstance.
```

josmperez replied:
Fiona, I agree with your phrasing here very much. The phrase "[Grafana Writers' Toolkit] recommends" is a powerful oe, and as I think I've made clear we probably shouldn't be throwing our weight behind the SLB spec that we don't use in any of our docs or writer's toolkit. But I think it's good that we give people the option to use it whenever it can add clarity to the review process.

imatwawana commented on docs/sources/write/style-guide/style-conventions/index.md:
````diff
@@ -219,3 +219,23 @@ On the published page, this warning renders as follows:
+When you write in active voice, you identify the subject of the sentence and the action that the subject performs. For example, “John drove the car” is active voice because it is clear that John (the subject) performed an action (drove). The passive voice variation is “The car was driven by John.”
+```
+
+The output is the same in both cases. However, the first is easier to review and edit and is less subject to the screen and text editor settings of each contributor.
````
@jdbaldry - I can't figure out what prettier wants me to do here.

jdbaldry replied:
I've updated the suggestion and committed it.

josmperez reviewed (approved):
LGTM with the new softening of recommendation language.

jdbaldry commented on docs/sources/write/style-guide/style-conventions/index.md:
```diff
@@ -219,3 +219,25 @@ On the published page, this warning renders as follows:
+
+The [Semantic Line Breaks organization](https://sembr.org/) suggests adding line breaks in your writing. We find this can help with reviews.
+
+Note that we don't require that this suggestion be followed. We share the Semantic Line Breaks rules so that you can discuss with your review team and adapt to your circumstance.
```
nit: Not using semantic line breaks in the PR about semantic line breaks :P

imatwawana replied:
```suggestion
Note that we don't require that this suggestion be followed. 
We share the Semantic Line Breaks rules so that you can discuss with your review team and adapt to your circumstance.
```

jdbaldry reviewed (approved):
I'm strongly in favor of this guidance as it significantly improves git diffs in the GitHub UI which is solely line based.

imatwawana commented on docs/sources/write/style-guide/style-conventions/index.md:
````diff
@@ -219,3 +219,25 @@ On the published page, this warning renders as follows:
+When you write in active voice, you identify the subject of the sentence and the action that the subject performs. For example, “John drove the car” is active voice because it is clear that John (the subject) performed an action (drove). The passive voice variation is “The car was driven by John.”
+```
+
+The output is the same in both cases. However, the first is easier to review and edit and is less subject to the screen and text editor settings of each contributor.
````
```suggestion
The output is the same in both cases. 
However, the first is easier to review and edit and is less subject to the screen and text editor settings of each contributor.
```

imatwawana commented on docs/sources/write/style-guide/style-conventions/index.md:
```diff
@@ -219,3 +219,25 @@ On the published page, this warning renders as follows:
+
+## Semantic line breaks
+
+The [Semantic Line Breaks organization](https://sembr.org/) suggests adding line breaks in your writing. We find this can help with reviews.
```
```suggestion
The [Semantic Line Breaks organization](https://sembr.org/) suggests adding line breaks in your writing. 
We find this can help with reviews.
```
